vcd_cli/vdc.py

```diff
@@ -117,12 +131,153 @@ def use(ctx, name):
                         ctx.obj['profiles'].set('vdc_href', str(v.href))
                         message = 'now using org: \'%s\', vdc: \'%s\'.' % \
                                   (in_use_org_name, name)
-                        stdout({'org': in_use_org_name, 'vdc': vdc},
-                               ctx,
-                               message)
+                        stdout({
+                            'org': in_use_org_name,
+                            'vdc': vdc
+                        }, ctx, message)
                         return
         raise Exception('not found')
     except Exception as e:
         stderr(e, ctx)
 
 
+@vdc.command(short_help='create a virtual datacenter')
+@click.pass_context
+@click.argument('name', metavar='<name>', required=True)
+@click.option(
+    '-p',
+    '--provider-vdc',
+    'pvdc_name',
+    required=True,
+    metavar='[provider-vdc]',
+    help='Provider VDC')
+@click.option(
+    '-n',
+    '--network-pool',
+    'network_pool_name',
+    required=False,
+    metavar='[network-pool]',
+    help='Network pool')
+@click.option(
+    'allocation_model',
+    '-a',
+    '--allocation-model',
+    type=click.Choice(['AllocationVApp', 'AllocationPool', 'ReservationPool']),
+    required=False,
+    default='AllocationVApp',
+    metavar='<allocation-model>',
+    help='Allocation model.')
+@click.option(
+    '-s',
+    '--storage-profile',
+    'pvdcsp_name',
+    default='*',
+    required=False,
+    metavar='[provider-vdc-storage-profile]',
+    help='Provider VDC Storage Profile.')
+@click.option(
+    '--storage-profile-limit',
+    'pvdcsp_limit',
+    default=0,
+    required=False,
+    metavar='[provider-vdc-storage-profile-limit]',
+    help='Provider VDC Storage Profile limit, 0 means unlimited.')
+@click.option(
+    '-D',
+    '--description',
+    required=False,
+    default='',
+    metavar='[description]',
+    help='description.')
+@click.option(
+    '--cpu-allocated',
+    required=False,
+    default=0,
+    metavar='<cpu-allocated>',
+    type=click.INT,
+    help='Capacity that is commited to be available.')
+@click.option(
+    '--cpu-limit',
+    required=False,
+    default=0,
+    metavar='<cpu-limit>',
+    type=click.INT,
+    help='Capacity limit relative to the value specified for Allocation.')
+def create(ctx, name, pvdc_name, network_pool_name, allocation_model,
+           pvdcsp_name, pvdcsp_limit, description, cpu_allocated, cpu_limit):
```

Comment:
We should provide support to add multiple storage profiles to an ovdc. Please track a separate Jira task as this change set is already big enough

Reply:
I already did, see VCDA-307 and VCDA-308